base_layer/core/src/proof_of_work/lwma.rs

```diff
@@ -0,0 +1,142 @@
+// LWMA-1 for BTC & Zcash clones
+// Copyright (c) 2017-2019 The Bitcoin Gold developers, Zawy, iamstenman (Microbitcoin)
+// MIT License
+// Algorithm by Zawy, a modification of WT-144 by Tom Harding
+// For updates see
+// https://github.com/zawy12/difficulty-algorithms/issues/3#issuecomment-442129791
+// Do not use Zcash's / Digishield's method of ignoring the ~6 most recent
+// timestamps via the median past timestamp (MTP of 11).
+//  FTL should be lowered to about N*T/20.
+//  FTL in BTC clones is MAX_FUTURE_BLOCK_TIME in chain.h.
+//  FTL in Ignition, Numus, and others can be found in main.h as DRIFT.
+//  FTL in Zcash & Dash clones need to change the 2*60*60 here:
+//  if (block.GetBlockTime() > nAdjustedTime + 2 * 60 * 60)
+//  which is around line 3700 in main.cpp in ZEC and validation.cpp in Dash
+//  If your coin uses median network time instead of node's time, the "revert to
+//  node time" rule (70 minutes in BCH, ZEC, & BTC) should be reduced to FTL/2
+//  to prevent 33% Sybil attack that can manipulate difficulty via timestamps. See:
+// https://github.com/zcash/zcash/issues/4021
+
+use crate::proof_of_work::difficulty::{Difficulty, DifficultyAjustment};
+use std::{cmp, collections::VecDeque};
+
+const TARGET: u64 = 60; // target time of 1 minute
+const BLOCK_WINDOW: u64 = 150; // we should cap the amount of entries int he VecDeques to this number.
+
+pub struct LWMA {
+    timestamps: VecDeque<u64>,
+    cumulative_difficulties: VecDeque<Difficulty>,
+}
+
+impl LWMA {
+    pub fn new() -> LWMA {
+        LWMA {
+            timestamps: VecDeque::new(),
+            cumulative_difficulties: VecDeque::new(),
+        }
+    }
+}
+
+impl DifficultyAjustment for LWMA {
+    fn add(&mut self, timestamp: u64, cum_difficulty: Difficulty) {
+        self.timestamps.push_back(timestamp);
+        self.cumulative_difficulties.push_back(cum_difficulty);
```

Comment:
Are you adding cumulative difficulties? Or target difficulties; because your test suggest the latter.

Reply:
Its target, previous version used cumulative. Forgot to rename the var after the refactor.